src/writ/transform/svdframe.py

```diff
@@ -0,0 +1,138 @@
+"""Provides a transform that rotates all molecules into their SVD frame.
+
+The RotateToSVDFrame class provides this functionality.
+"""
+
+from typing import (
+    Iterable,
+    Iterator,
+    TypeVar,
+    Annotated,
+    Sequence,
+    Literal,
+    Tuple,
+)
+from typing_extensions import TypeGuard
+
+import numpy as np
+import numpy.typing as npt
+
+A = TypeVar("A")
+
+DType = TypeVar("DType", bound=np.generic)
+
+ArrayNx3 = Annotated[npt.NDArray[DType], Literal["N", 3]]
+ArrayMxNx3 = Annotated[npt.NDArray[DType], Literal["M", "N", 3]]
+
+
+# TODO: write tests!
+class RotateToSVDFrame(Iterable[Sequence[A]]):
+    """Provides an iterable that rotates a source configuration into its SVD-frame.
+
+    This class returns rotated coords and forces by modifying the respective entries
+    of the iterates in-place. The positions of the coordinates and forces in iterate
+    can be specified at initialization.
+    """
+
+    def __init__(
+        self,
+        source: Iterable[Sequence[A]],
+        coords_idx: int = 0,
```

Comment:
maybe change the "idx" vars to "index", only to be consistent with the rest of the code base?